underscore.js

```diff
@@ -1115,6 +1115,15 @@
     if (a === b) return a !== 0 || 1 / a === 1 / b;
     // A strict comparison is necessary because `null == undefined`.
     if (a == null || b == null) return a === b;
+    // `NaN`s are equivalent, but non-reflexive.
+    if (a !== a) return b !== b;
+    // Exhaust primitive checks
+    var type = typeof a;
+    if (type !== 'function' && type !== 'object' && typeof b !== 'object') return false;
```

Comment:
I'm unsure if it's needed:

| type a | type b | equals |
| -------- |  -------- |  -------- | 
| primative | primative | primitive checks |
| primative | function | false |
| primative  | object |  deepEq |
| function | primative | false |
| function | function | deepEq |
| function | object | deepEq |
| object | primative | deepEq |
| object | function | false |
| object | object | deepEq |

There's a difference between `(function, object)` and `(object, function)`, but I think both those could be handled by primitive checks only? It's was just easier to check for `type === 'function'` for the `(function, function)` case (Safari regexs).